File: problem32.cpp
#include <bits/stdc++.h>
using namespace std;

set<long int> s4, s5, s6, s7, s8, s9;

bool ispandigit(string temp)
{
	int i, a[10] = {0}, flag;

	for(i=0; i<temp.size(); i++)
	{
		a[temp[i] - 48]++;
	}

	if(a[0] == 0)
		flag = 1;
	else
		flag = 0;

	i = 1;
	while(a[i] == 1)
		i++;

	while(i<10)
	{
		if(a[i]>0)
			flag = 0;
		i++;
	}

	if(flag == 1)
		return true;
	else
		return false;
}

void pre()
{
	int i, j, pro, start, end;
	string temp, st_i, st_j, st_pro;

	for(i=2; i<100; i++)
	{
		//start = (i>9) ? 123 : 1234;
		end = 10000/i;

		for(j=2; j<end; j++)
		{
			pro = i*j;
			temp = to_string(i) + to_string(j) + to_string(pro);
			if(ispandigit(temp))
			{
				if(temp.size() == 4)
					s4.insert(pro);
				if(temp.size() == 5)
					s5.insert(pro);
				if(temp.size() == 6)
					s6.insert(pro);
				if(temp.size() == 7)
					s7.insert(pro);
				if(temp.size() == 8)
					s8.insert(pro);
				if(temp.size() == 9)
					s9.insert(pro);
			}
		}
	}
}

int main()
{
	pre();
	set<long int>::iterator it;

	//for(it=s4.begin(); it!=s4.end(); it++)
		//cout << (*it) << endl;

	int n, ans[10] = {0};

	for(it=s4.begin(); it!=s4.end(); it++)
		ans[4] += (*it);
	for(it=s5.begin(); it!=s5.end(); it++)
		ans[5] += (*it);
	for(it=s6.begin(); it!=s6.end(); it++)
		ans[6] += (*it);
	for(it=s7.begin(); it!=s7.end(); it++)
		ans[7] += (*it);
	for(it=s8.begin(); it!=s8.end(); it++)
		ans[8] += (*it);
	for(it=s9.begin(); it!=s9.end(); it++)
		ans[9] += (*it);
	
	cin >> n;
	cout << ans[n] << endl;
	return 0;
}

File: problem32.py
import sys 

pan4 = []
pan5 = []
pan6 = []
pan7 = []
pan8 = []
pan9 = []

def ispandigit(n):
	flag = 1
	a = [0]
	a = a*11

	for i in range(0, len(n)):
		a[int(n[i])]+=1
	
	if a[0] == 0:
		j = 1
		while a[j] == 1 and j<10:
			j+=1
		while j<10:
			if a[j] > 0:
				flag = 0
			j+=1
	else:
		flag = 0

	if flag == 1:
		return True
	else:
		return False

def pre():
	for i in range(2, 100):
		for j in range(2, 10000//i):
			pro = i*j
			temp = str(i)+str(j)+str(pro)

			if ispandigit(temp):
				if len(temp) == 4:
					pan4.append(pro)
				if len(temp) == 5:
					pan5.append(pro)
				if len(temp) == 6:
					pan6.append(pro)
				if len(temp) == 7:
					pan7.append(pro)
				if len(temp) == 8:
					pan8.append(pro)
				if len(temp) == 9:
					pan9.append(pro)
				

def main():
	pre()
	#print(pan4, pan5, pan6, pan7, pan8, pan9)

	n = int(input())
	ans = 0

	if n == 4:
		for value in set(pan4):
			ans += value
	if n == 5:
		for value in set(pan5):
			ans += value
	if n == 6:
		for value in set(pan6):
			ans += value
	if n == 7:
		for value in set(pan7):
			ans += value
	if n == 8:
		for value in set(pan8):
			ans += value
	if n == 9:
		for value in set(pan9):
			ans += value				

	print(ans)


if __name__ == '__main__':
	main()
